crates/torii/core/src/engine.rs

```diff
@@ -268,11 +186,11 @@ impl<P: Provider + Sync> Engine<P> {
                                 } else {
                                     // If the block is pending, we assume the block number is the
                                     // latest + 1
-                                    to + 1
+                                    latest_block_number + 1
                                 }
                             }
 
-                            _ => to + 1,
+                            _ => latest_block_number + 1,
                         }
                     }
                 };
```

Comment:
afaik all events (apart from events from pending block) contains the block number, so we can assume if its not present then its an event from pending block. so we dont need to call `get_transction_receipt` for every event in pending block, wdyt?

Reply:
yes nice catch.